### PR TITLE
Feature/kintsugi support

### PR DESCRIPTION
**Pull request by bnsports**

### CODE REVIEW

File: fearless/Common/DataProvider/Adapter/AccountInfoSubscriptionAdapter.swift [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    init(
+        walletLocalSubscriptionFactory: WalletLocalSubscriptionFactoryProtocol,
+        selectedMetaAccount: MetaAccountModel
+    ) {
+        self.walletLocalSubscriptionFactory = walletLocalSubscriptionFactory
+        self.selectedMetaAccount = selectedMetaAccount
+    }
+
+    func reset() {
+        if let providers = ormlAccountInfoProviders {
```
Comment (posplaw) [inline fearless/Common/DataProvider/Adapter/AccountInfoSubscriptionAdapter.swift:35]: Why not `ormlAccountInfoProviders?.foreach { $0.removeObserver(self) }`

File: fearless/Common/DataProvider/Adapter/AccountInfoSubscriptionAdapter.swift [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        var ormlProviders: [AnyDataProvider<DecodedOrmlAccountInfo>] = []
+        var defaultProviders: [AnyDataProvider<DecodedAccountInfo>] = []
+
+        if chain.chainId.isOrml {
+            if let provider = subscribeToOrmlAccountInfoProvider(for: accountId, chain: chain) {
+                ormlProviders.append(provider)
+            }
+        }
+
+        if !chain.chainId.isOrml {
```
Comment (posplaw) [inline fearless/Common/DataProvider/Adapter/AccountInfoSubscriptionAdapter.swift:62]: It looks like it's definitely `else` case for previous `if`, please fix

File: fearless/Common/DataProvider/Adapter/AccountInfoSubscriptionAdapter.swift [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        var defaultProviders: [AnyDataProvider<DecodedAccountInfo>] = []
+
+        chains.forEach { chain in
+            if chain.chainId.isOrml, let accountId = selectedMetaAccount.fetch(for: chain.accountRequest())?.accountId {
+                if let provider = subscribeToOrmlAccountInfoProvider(for: accountId, chain: chain) {
+                    ormlProviders.append(provider)
+                }
+            }
+
+            if !chain.chainId.isOrml, let accountId = selectedMetaAccount.fetch(for: chain.accountRequest())?.accountId {
```
Comment (posplaw) [inline fearless/Common/DataProvider/Adapter/AccountInfoSubscriptionAdapter.swift:87]: Again, `else` instead. And you can definitely unwrap `accountId` before `if/else`

File: fearless/Common/DataProvider/Subscription/WalletLocalStorageSubscriber.swift [hunk 4]
```diff
@@ -57,6 +63,64 @@ extension WalletLocalStorageSubscriber {
...
+
+        let updateClosure = { [weak self] (changes: [DataProviderChange<DecodedOrmlAccountInfo>]) in
+            guard let ormlAccountInfo = changes.reduceToLastChange()?.item else {
+                return
+            }
+            let accountInfo = AccountInfo(
+                nonce: 0,
+                consumers: 0,
+                providers: 0,
+                data: AccountData(
```
Comment (posplaw) [inline fearless/Common/DataProvider/Subscription/WalletLocalStorageSubscriber.swift:86]: Maybe add additional constructor for `AccountData` which accepts `OrmlAccountInfo`?

File: fearless/Common/Extension/Wallet/TransactionHistoryItem+Wallet.swift [hunk 5]
```diff
@@ -31,7 +31,9 @@ extension TransactionHistoryItem {
         }
 
         let callPath = CallCodingPath.transfer
-        let callArgs = TransferCall(dest: .accoundId(receiverAccountId), value: amount)
+
+        // Not used anymore
```
Comment (posplaw) [inline fearless/Common/Extension/Wallet/TransactionHistoryItem+Wallet.swift:35]: ??

File: fearless/Common/Model/ChainRegistry/ChainModel.swift [hunk 6]
```diff
@@ -1,6 +1,12 @@
 import Foundation
 import RobinHood
 
+extension ChainModel.Id {
+    var isOrml: Bool {
+        self == "9af9a64e6e4da8e3073901c3ff0cc4c3aad9563786d89daf6ad820b6e14a0b8b"
```
Comment (posplaw) [inline fearless/Common/Model/ChainRegistry/ChainModel.swift:6]: Consider moving to some global `Constants.swift`, and it's a candidate for remote config.json in utils

File: fearless/Common/Network/Wallet/WalletNetworkOperationFactory+Protocol.swift [hunk 7]
```diff
@@ -51,7 +51,8 @@ extension WalletNetworkOperationFactory: WalletNetworkOperationFactoryProtocol {
         let compoundReceiver = createAccountInfoFetchOperation(receiver)
 
         let builderClosure: ExtrinsicBuilderClosure = { builder in
-            let call = SubstrateCallFactory().transfer(to: receiver, amount: amount)
+            // NOT USED
```
Comment (posplaw) [inline fearless/Common/Network/Wallet/WalletNetworkOperationFactory+Protocol.swift:54]: ??

File: fearless/Common/Substrate/Calls/TransferCall.swift [hunk 8]
```diff
@@ -2,7 +2,88 @@ import Foundation
 import FearlessUtils
 import BigInt
 
+enum TokenSymbol: String {
```
Comment (posplaw) [inline fearless/Common/Substrate/Calls/TransferCall.swift:5]: I would discuss this approach separately

File: fearless/Modules/Staking/StakingAmount/StakingAmountInteractor.swift [hunk 9]
```diff
@@ -92,16 +92,20 @@ extension StakingAmountInteractor: StakingAmountInteractorInputProtocol, Runtime
...
-            runtimeCodingService: runtimeService,
-            operationManager: operationManager
-        ) { [weak self] (result: Result<BigUInt, Error>) in
-            switch result {
-            case let .success(amount):
-                self?.presenter.didReceive(minimalBalance: amount)
-            case let .failure(error):
-                self?.presenter.didReceive(error: error)
+        if chain.isOrml {
+            presenter.didReceive(minimalBalance: BigUInt.zero)
```
Comment (posplaw) [inline fearless/Modules/Staking/StakingAmount/StakingAmountInteractor.swift:96]: `0` not being accepted? I believe `BigUInt` conforms to ExpressedByIntegerLiteral

File: fearless/Modules/Staking/StakingRedeem/StakingRedeemInteractor.swift [hunk 10]
```diff
@@ -174,12 +174,16 @@ extension StakingRedeemInteractor: StakingRedeemInteractorInputProtocol {
...
-        ) { [weak self] (result: Result<BigUInt, Error>) in
-            self?.presenter.didReceiveExistentialDeposit(result: result)
+        if chain.isOrml {
+            presenter?.didReceiveExistentialDeposit(result: .success(BigUInt.zero))
+        } else {
+            fetchConstant(
+                for: .existentialDeposit,
+                runtimeCodingService: runtimeService,
+                operationManager: operationManager
+            ) { [weak self] (result: Result<BigUInt, Error>) in
```
Comment (posplaw) [inline fearless/Modules/Staking/StakingRedeem/StakingRedeemInteractor.swift:184]: Is `(result: Result<BigUInt, Error>)` really required to be declared?

**Review by posplaw — CHANGES_REQUESTED** [review]

File: fearless/Modules/Crowdloan/CrowdloanContribution/CrowdloanContributionInteractor.swift [hunk 11]
```diff
@@ -69,12 +70,16 @@ class CrowdloanContributionInteractor: CrowdloanContributionInteractorInputProto
...
         }
 
-        fetchConstant(
-            for: .existentialDeposit,
-            runtimeCodingService: runtimeService,
-            operationManager: operationManager
-        ) { [weak self] (result: Result<BigUInt, Error>) in
-            self?.presenter.didReceiveMinimumBalance(result: result)
+        if chainAsset.chain.isOrml {
+            presenter?.didReceiveMinimumBalance(result: .success(BigUInt.zero))
```
Comment (Viscount88) [inline fearless/Modules/Crowdloan/CrowdloanContribution/CrowdloanContributionInteractor.swift:74]: It's not zero in result?

File: fearless/Modules/NewWallet/WalletSend/WalletSendInteractor.swift [hunk 12]
```diff
@@ -55,12 +55,16 @@ final class WalletSendInteractor: RuntimeConstantFetching {
             self?.presenter?.didReceiveBlockDuration(result: result)
         }
 
-        fetchConstant(
-            for: .existentialDeposit,
-            runtimeCodingService: runtimeService,
-            operationManager: operationManager
-        ) { [weak self] (result: Result<BigUInt, Error>) in
-            self?.presenter?.didReceiveMinimumBalance(result: result)
+        if chain.isOrml {
```
Comment (Viscount88) [inline fearless/Modules/NewWallet/WalletSend/WalletSendInteractor.swift:58]: I already found same logic in several modules, mb we can use some common handler for all of them.

**Review by Viscount88 — CHANGES_REQUESTED** [review]

File: fearless/Modules/Crowdloan/CrowdloanContribution/CrowdloanContributionInteractor.swift (hunk 11 above)
Comment (posplaw) [reply fearless/Modules/Crowdloan/CrowdloanContribution/CrowdloanContributionInteractor.swift:74]: Yeah should work conforming to ExpressibleByIntegerLiteral or something like that

File: fearless/Common/Configs/ApplicationConfigs.swift [hunk 13]
```diff
@@ -118,7 +118,7 @@ extension ApplicationConfig: ApplicationConfigProtocol {
     // MARK: - GitHub
 
     var chainListURL: URL? {
-        GitHubUrl.url(suffix: "chains/chains.json")
+        GitHubUrl.url(suffix: "chains/chains_dev.json")
```
Comment (Viscount88) [inline fearless/Common/Configs/ApplicationConfigs.swift]: Don't forgive to return chains.json

**Review by Viscount88 — APPROVED** [review]

File: fearless/Common/Network/Wallet/WalletNetworkOperationFactory+Protocol.swift (hunk 7 above)
Comment (bnsports) [reply fearless/Common/Network/Wallet/WalletNetworkOperationFactory+Protocol.swift]: code from common wallet (capital). need to remove all capital-based code from app

**Review by Viscount88 — APPROVED** [review]

**Pull request closed without merging**